### PR TITLE
expiration_start_work.py: Honor pull requests

### PR DESCRIPTION
**Pull request by pinkiebell**
As promised my lovely @gitcoinbot :*

Related #1769

### CODE REVIEW

File: app/git/utils.py [hunk 1]
```diff
@@ -475,6 +475,9 @@ def get_interested_actions(github_url, username, email=''):
         elif 'user' in action:
             gh_user = action['user']['login']
 
+        if gh_user and gh_user == username and action['event'] in activity_event_types:
+            actions_by_interested_party.append(action)
+
         if action['event'] == 'cross-referenced':
```
Comment (pinkiebell) [inline app/git/utils.py]: Actually, we shouldn't fetch further at this point.
It's likely that we fetch things and (prs) from other issues, that gives us false-positives 🤔

**Comment by owocki** [comment]
> As promised my lovely @gitcoinbot :*

lol. ...


reviewing now!

**Comment by owocki** [comment]
two questions:

1. what does this do that https://github.com/gitcoinco/web/pull/458 doesnt?  sorry, having a hard time following the diff's in the code
2. how did you test this?

File: app/git/utils.py (hunk 1 above)
Comment (owocki) [reply app/git/utils.py]: wouldnt fetching more things / prs from other issues give us less false-negatives (more PRs, more chances we could find a place where the contributor commented)

File: app/marketing/management/commands/expiration_start_work.py [hunk 2]
```diff
@@ -99,28 +99,45 @@ def handle(self, *args, **options):
...
-                            delta_now_vs_last_action = timezone.now() - snooze_time - last_action_by_user
-                            last_heard_from_user_days = delta_now_vs_last_action.days
-
-                            # decide action params
-                            should_warn_user = last_heard_from_user_days >= num_days_back_to_warn
-                            should_delete_interest = last_heard_from_user_days >= num_days_back_to_delete_interest
-                            should_ignore = last_heard_from_user_days >= num_days_back_to_ignore_bc_mods_got_it
-
-                            print(f"- its been {last_heard_from_user_days} days since we heard from the user")
+                            action_times = []
```
Comment (owocki) [inline app/marketing/management/commands/expiration_start_work.py]: whyd you remove the `datetime.strptime(action['created_at'], '%Y-%m-%dT%H:%M:%SZ') ` from this list?

Comment (owocki) [reply app/marketing/management/commands/expiration_start_work.py]: ah nvm, isee it was moved below

Comment (pinkiebell) [reply app/marketing/management/commands/expiration_start_work.py]: This was the first bug, see variable assignment. 😄

Comment (pinkiebell) [reply app/marketing/management/commands/expiration_start_work.py]: Ah, sry that was no bug here. I had my mind on the wrong file

File: app/git/utils.py (hunk 1 above)
Comment (pinkiebell) [reply app/git/utils.py]: That was the bug I'm talking about first (variable assignment/overwrite) - the gh_user.

With false-negatives I mean if we fetch further we very likely fetch PRs unrelated to the issue in question.
If we just do one loop we should get all the events we need, at least if the user references the issue in the commits/pull request(s). That may also include false-negatives, but that would be rare.

**Comment by pinkiebell** [comment]
> two questions:
> 
> 1. what does this do that #458 doesnt?  sorry, having a hard time following the diff's in the code
We bail out early if we recognize a pull request (we do not check the time in this case).
> 2. how did you test this?
Locally with advanced loggin 👍

**Comment by owocki** [comment]
> We bail out early if we recognize a pull request (we do not check the time in this case).

makes sense

> Locally with advanced loggin 👍

got it; i have not tested this.. did you create a issue / PR on github to test this?  if so mind passing along the link so i can test it too?

**Comment by pinkiebell** [comment]
> > We bail out early if we recognize a pull request (we do not check the time in this case).
> 
> makes sense
> 
> > Locally with advanced loggin 👍
> 
> got it; i have not tested this.. did you create a issue / PR on github to test this? if so mind passing along the link so i can test it too?

I tested this with the mainnet issues, I just commented out the 'if settings.DEBUG: return' things .

**Comment by pinkiebell** [comment]
@owocki 
btw,
```
should_continue_loop = len(actions) == 100
```
we should change that to `!= 0` because I had seen timeline results with multiple pages but returning < 100 per request. I don't know if that is still the case though

**Comment by mbeacom** [comment]
:star: PR built and available in a preview environment **gitcoinco-web-pr-2176** [here](http://web.jx-gitcoinco-web-pr-2176.ci.gitcoin.co)

**Pull request closed without merging**